test/suites/test_socket_fd.py

```diff
@@ -0,0 +1,175 @@
+        sock.connect((self.srv.host, self.tcp_port))
+        return sock
+
+    def test_01_socket_fd_connect(self):
+        sock = self._get_tt_sock()
+        conn = tarantool.connect(host=None, port=sock.fileno())
+        sock.detach()
+        try:
+            self.assertSequenceEqual(conn.eval(self.EVAL_USER), ["test"])
+        finally:
+            conn.close()
+
+    def test_02_socket_fd_re_auth(self):
+        sock = self._get_tt_sock()
+        conn = tarantool.connect(host=None, port=sock.fileno(), user="guest")
+        sock.detach()
+        try:
+            self.assertSequenceEqual(conn.eval(self.EVAL_USER), ["guest"])
+        finally:
+            conn.close()
+
+    def test_03_socket_fd_mesh(self):
+        sock = self._get_tt_sock()
+        pool = tarantool.ConnectionPool(
+            addrs=[{'host': None, 'port': sock.fileno()}]
+        )
+        sock.detach()
+        try:
+            self.assertSequenceEqual(pool.eval(self.EVAL_USER, mode=tarantool.Mode.ANY), ["test"])
+        finally:
+            pool.close()
+
+    def test_04_socket_fd_pool(self):
+        sock = self._get_tt_sock()
+        mesh = tarantool.MeshConnection(
+            host=None, port=sock.fileno()
+        )
+        sock.detach()
+        try:
+            self.assertSequenceEqual(mesh.eval(self.EVAL_USER), ["test"])
+        finally:
+            mesh.close()
+
+    @unittest.skipIf(sys.platform.startswith("win"),
+                     "Skip on Windows since it uses remote server")
+    def test_05_tarantool_made_socket(self):
+        python_exe = find_python()
+        cwd = os.getcwd()
+        side_script_path = os.path.join(cwd, "test", "suites", "sidecar.py")
+
+        # pylint:disable=C0209
+        ret_code, err = self.srv.admin("""
+            local socket = require('socket')
+            local popen = require('popen')
+            local os = require('os')
+            local s1, s2 = socket.socketpair('AF_UNIX', 'SOCK_STREAM', 0)
+
+            --[[ Tell sidecar which fd use to connect. --]]
+            os.setenv('SOCKET_FD', tostring(s2:fd()))
+
+            --[[ Tell sidecar where find `tarantool` module. --]]
+            os.setenv('PYTHONPATH', (os.getenv('PYTHONPATH') or '') .. ':' .. '%s')
+
+            box.session.new({
+                type = 'binary',
+                fd = s1:fd(),
+                user = 'test',
+            })
+            s1:detach()
+
+            local ph, err = popen.new({'%s', '%s'}, {
+                stdout = popen.opts.PIPE,
+                stderr = popen.opts.PIPE,
+                inherit_fds = {s2:fd()},
+            })
+
+            if err ~= nil then
+                return 1, err
+            end
+
+            ph:wait()
+
+            local status_code = ph:info().status.exit_code
+            local stderr = ph:read({stderr=true}):rstrip()
+            return status_code, stderr
+        """ % (cwd, python_exe, side_script_path))
+        if err is not None:
```

Comment:
You can use the third argument: `assertEqual(first, second, msg=None)`

Reply:
Changed.